### ISSUE TITLE
}} inside hfun arguments

### ISSUE THREAD

**Issue by fredrikekre** [tlienart/Franklin.jl#573]
```
$ cat utils.jl 
hfun_identity(arg) = join(arg, ' ')

$ cat index.html 
{{ identity {{ foo }} }}

$ cat __site/index.html 
{{ foo
```

I would expect this to do either of:

 1. Call `hfun_identity` with `arg = ["{{", "foo", "}}"]`
 2. Call `hfun_identity` with the result of `hfun_foo`

**Comment by tlienart** [comment]
Yes, at the moment nesting is not supported (as you can see); to support it would mainly require parsing double braces keeping track of balance ({{ vs }}) I actually did not do that on purpose as I thought it was better to avoid this kind of syntax in the markdown & encouragé people to do this kind of stuff inside the definition of a hfun as it's a bit of a pain to read with all these braces.

But maybe that it would be nice to have occasionally. Could you give me an example of what you were trying to do with this? Thanks!

**Comment by fredrikekre** [comment]
I wanted something like
```
{{ define title="#{{ fill fd_tag }}" }}

{{ fill title }}
```
but yea, I easily worked around this by doing
```
{{ definetagtitle }}

{{ fill title }}
```
so it is no big deal. Perhaps document the argument parsing to  `hfun`s is sufficient here.
